log/context.go

```diff
@@ -21,16 +22,28 @@ const (
 // All zero value fields will be ignored and only non-zero values will be
 // attached.
 type AttachArgs struct {
-	TraceID uint64
+	// The trace id of the span.
+	//
+	// If StringTraceID is set to true,
+	// it will be logged as string instead of int.
+	// The conversion to string will make it slightly slower,
+	// but avoids troubles when some of the log ingesters cannot deal with uint64
+	// numbers larger than max signed int64.
+	TraceID       uint64
+	StringTraceID bool
```

Comment:
Maybe we should just string-ify it always?

Reply:
That was my first version. But in case in the future our log ingester fixed the issue and can handle uint64 fine, we would want to avoid the extra string conversion to save some performance, so leave it flexible seems slightly more valuable?

Reply:
I imagine the performance hit isn't that much?  I also tend to err on the side of making things simpler when we don't need the flexibility and adding flexibility later if we find we need it.

Reply:
It's ~55ns per request (since this is usually done once per request):

```
$ go test -bench . -benchmem
goos: linux
goarch: amd64
BenchmarkFormatUint/baseline-12                 256251235                4.50 ns/op            0 B/op          0 allocs/op
BenchmarkFormatUint/format-uint-12              19179976                59.2 ns/op            31 B/op          1 allocs/op
PASS
$ cat bench_test.go 
package main

import (
        "math/rand"
        "strconv"
        "testing"
)

func BenchmarkFormatUint(b *testing.B) {
        r := rand.New(rand.NewSource(0))

        r.Seed(0)
        b.Run(
                "baseline",
                func(b *testing.B) {
                        for i := 0; i < b.N; i++ {
                                r.Uint64()
                        }
                },
        )

        r.Seed(0)
        b.Run(
                "format-uint",
                func(b *testing.B) {
                        for i := 0; i < b.N; i++ {
                                strconv.FormatUint(r.Uint64(), 10)
                        }
                },
        )
}
```

is it "much" is up to your definition :) I don't feel strongly either way either. will remove it.